File: challenges/codeeval/medium/005/redexel.py
#Alexander Botero 

import sys

with open(sys.argv[1], 'r') as input:
    testing = input.read().strip().splitlines()

for test in testing:
    secuencia = [int(i) for i in test.split()]
    seq_length = len(secuencia)

    x, y = 0, 1
    while secuencia[x] != secuencia[y]:
        x = (x + 1) % seq_length
        y = (y + 2) % seq_length

    begin = 0
    y, x = abs(y - x), 0
    while secuencia[x] != secuencia[y]:
        x = (x + 1) % seq_length
        y = (y + 1) % seq_length
        begin += 1

    length, y = 1, x + 1
    while secuencia[x] != secuencia[y]:
        y += 1
        length += 1

    print ' '.join(str(i) for i in secuencia[begin:begin + length])

File: challenges/codeeval/easy/091/redexel.rb
# Alexander Botero
def sort(list)
  for i in 1...list.length
    a = i
    while a > 0 && list[a - 1] > list[a]
      list[a - 1], list[a] = list[a], list[a - 1]
      a -= 1
    end
  end

  list
end

File.open(ARGV[0]).each_line do |line|
  numbers = line.split(' ').map(&:to_f)
  puts sort(numbers).map { |n| '%0.3f' % n  }.join(' ')
end
